src/intelligence_layer/evaluation/dataset/studio_dataset_repository.py

```diff
@@ -0,0 +1,195 @@
+            labels: A list of labels for filtering. Defaults to an empty list.
+            metadata: A dict for additional information about the dataset. Defaults to an empty dict.
+
+        Returns:
+           :class:`Dataset`
+           :exception: Raises DataInternalError exception when the client fails.
+        """
+        if id is not None:
+            raise NotImplementedError(
+                "Custom dataset IDs are not supported by the StudioDataRepository"
+            )
+        source_data_list = [example.model_dump_json() for example in examples]
+        remote_dataset = self.data_client.create_dataset(
+            repository_id=self.repository_id,
+            dataset=DatasetCreate(
+                source_data="\n".join(source_data_list).encode(),
+                name=dataset_name,
+                labels=list(labels) if labels is not None else [],
+                total_datapoints=len(source_data_list),
+                metadata=metadata,
+            ),
+        )
+        return Dataset(
+            id=remote_dataset.dataset_id,
+            name=remote_dataset.name or "",
+            labels=set(remote_dataset.labels)
+            if remote_dataset.labels is not None
+            else set(),
+            metadata=remote_dataset.metadata or dict(),
+        )
+
+    def delete_dataset(self, dataset_id: str) -> None:
+        """Deletes a dataset identified by the given dataset ID.
+
+        Args:
+            dataset_id: Dataset ID of the dataset to delete.
+
+        Returns:
+            :exception: Raises DataInternalError exception when the client fails.
+        """
+        self.data_client.delete_dataset(
+            repository_id=self.repository_id, dataset_id=dataset_id
+        )
+
+    def dataset(self, dataset_id: str) -> Optional[Dataset]:
+        """Returns a dataset identified by the given dataset ID.
+
+        Args:
+            dataset_id: Dataset ID of the dataset to delete.
+
+        Returns:
+            :class:`Dataset` if it was not, `None` otherwise.
+            :exception: Raises DataInternalError exception when the client fails.
+        """
+        remote_dataset = self.data_client.get_dataset(
+            repository_id=self.repository_id, dataset_id=dataset_id
+        )
+        return Dataset(
+            id=remote_dataset.dataset_id,
+            name=remote_dataset.name or "",
+            labels=set(remote_dataset.labels)
+            if remote_dataset.labels is not None
+            else set(),
+            metadata=remote_dataset.metadata or dict(),
+        )
+
+    def datasets(self) -> Iterable[Dataset]:
+        """Returns all :class:`Dataset`s sorted by their ID.
+
+        Returns:
+            :class:`Sequence` of :class:`Dataset`s.
+            :exception: Raises DataInternalError exception when the client fails.
+        """
+        for remote_dataset in self.data_client.list_datasets(
+            repository_id=self.repository_id
+        ):
```

Comment:
This is not sorted i believe, so it should not be in the docstring that its sorted. I would have to look into why we did this, but it might be important that they are sorted